internal/addrs/provider_function.go

```diff
@@ -0,0 +1,26 @@
+package addrs
+
+import (
+	"fmt"
+)
+
+// ProviderFunction is the address of a provider defined function.
+type ProviderFunction struct {
+	referenceable
+	Name     string
+	Alias    string
```

Comment:
tiny nitpick: can we call this ProviderAlias or something, so that nobody thinks it's referencing a function alias?